consensus/state.go

```diff
@@ -2363,35 +2373,36 @@ func (cs *State) signAddVote(
 	msgType cmtproto.SignedMsgType,
 	hash []byte,
 	header types.PartSetHeader,
-) *types.Vote {
+) {
 	if cs.privValidator == nil { // the node does not have a key
-		return nil
+		return
 	}
 
 	if cs.privValidatorPubKey == nil {
 		// Vote won't be signed, but it's not critical.
 		cs.Logger.Error(fmt.Sprintf("signAddVote: %v", errPubKeyIsNotSet))
-		return nil
+		return
 	}
 
 	// If the node not in the validator set, do nothing.
 	if !cs.Validators.HasAddress(cs.privValidatorPubKey.Address()) {
-		return nil
+		return
 	}
 
 	// TODO: pass pubKey to signVote
 	vote, err := cs.signVote(msgType, hash, header)
 	if err != nil {
 		cs.Logger.Error("failed signing vote", "height", cs.Height, "round", cs.Round, "vote", vote, "err", err)
-		return nil
+		return
 	}
-	if !cs.state.ConsensusParams.ABCI.VoteExtensionsEnabled(vote.Height) {
-		// The signer will sign the extension, make sure to remove the data on the way out
-		vote.StripExtension()
+	hasExt := len(vote.ExtensionSignature) > 0
```

Comment:
I would leave them here for the time being, I prefer to be over-exhaustive... until this code has been used to some extent.
Just in case there is another path we didn't think of. Remember that the initial point of this code was ruling out possible inconsistencies with extension data when troubleshooting a real problem.